x/pki/handler_test.go

```diff
@@ -135,1703 +135,1983 @@ func Setup(t *testing.T) *TestSetup {
+			rootCertificate.SerialNumber = SerialNumber
+			setup.Keeper.SetUniqueCertificate(
+				setup.Ctx,
+				uniqueCertificate(rootCertificate.Subject, rootCertificate.SerialNumber),
+			)
+			setup.Keeper.AddApprovedCertificate(setup.Ctx, rootCertificate)
+
+			// propose second root certificate
+			_, err := ProposeX509RootCert(setup, setup.Trustee1, testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
+			require.NoError(t, err)
+
+			// check
+			certificate, _ := querySingleApprovedCertificate(setup, testconstants.RootSubject, testconstants.RootSubjectKeyID)
+			require.True(t, certificate.IsRoot)
+			require.Equal(t, testconstants.RootIssuer, certificate.Subject)
+			require.Equal(t, SerialNumber, certificate.SerialNumber)
+
+			proposedCertificate, _ := queryProposedCertificate(setup, testconstants.RootSubject, testconstants.RootSubjectKeyID)
+			require.Equal(t, testconstants.RootIssuer, proposedCertificate.Subject)
+			require.Equal(t, testconstants.RootSerialNumber, proposedCertificate.SerialNumber)
+
+			require.NotEqual(t, certificate.SerialNumber, proposedCertificate.SerialNumber)
+		})
+
+		t.Run("ForDifferentSerialNumberDifferentSigner", func(t *testing.T) {
+			setup := Setup(t)
+
+			// store root certificate with different serial number
+			rootCertificate := rootCertificate(testconstants.Address1)
+			rootCertificate.SerialNumber = SerialNumber
+			setup.Keeper.SetUniqueCertificate(
+				setup.Ctx,
+				uniqueCertificate(rootCertificate.Subject, rootCertificate.SerialNumber),
+			)
+			setup.Keeper.AddApprovedCertificate(setup.Ctx, rootCertificate)
+
+			// propose second root certificate
+			_, err := ProposeX509RootCert(setup, setup.Trustee1, testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
+			require.Error(t, err)
+			require.True(t, sdkerrors.ErrUnauthorized.Is(err))
+		})
+	})
+
+	t.Run("ProposeAddAndRejectX509RootCert", func(t *testing.T) {
+
+		t.Run("ByTrustee", func(t *testing.T) {
+			setup := Setup(t)
+
+			// propose x509 root certificate
+			_, err := ProposeX509RootCert(setup, setup.Trustee1, testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
+			require.NoError(t, err)
+
+			// reject x509 root certificate
+			_, err = RejectX509RootCert(setup, setup.Trustee1, testconstants.RootSubject, testconstants.RootSubjectKeyID, testconstants.Info)
+			require.NoError(t, err)
+
+			require.False(t, setup.Keeper.IsProposedCertificatePresent(setup.Ctx, testconstants.RootIssuer, testconstants.RootSerialNumber))
+
+			// check that unique certificate key is registered
+			require.False(t, setup.Keeper.IsUniqueCertificatePresent(
+				setup.Ctx, testconstants.RootIssuer, testconstants.RootSerialNumber))
+		})
+
+		t.Run("ByAnotherTrustee", func(t *testing.T) {
+			setup := Setup(t)
+
+			// propose x509 root certificate
+			proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(setup.Trustee1.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
```

Comment:
Why don't we use proposeAddX509RootCert function here?

Reply:
Didn't notice, fixed it